packages/calypso-e2e/src/lib/components/support-component.ts

```diff
@@ -85,69 +58,40 @@ export class SupportComponent {
 		await this.page.waitForSelector( selectors.closeButton, { state: 'hidden' } );
 	}
 
-	/**
-	 * Wait for and scroll to expose the Support card, present only on My Home.
-	 */
-	async showSupportCard(): Promise< void > {
-		const locator = this.page.locator( selectors.supportCard );
-		await Promise.all( [ locator.waitFor(), locator.scrollIntoViewIfNeeded() ] );
-	}
-
-	/* Result methods */
+	/* Results */
 
 	/**
-	 * Checks whether the Support popover/card is in a default state (ie. no search keyword).
-	 */
-	async defaultStateShown(): Promise< void > {
-		await this.page.waitForSelector( selectors.defaultResultsMessage );
-	}
-
-	/**
-	 * Given a selector, returns a Locator that match the given selector.
-	 * Prior to selecing the elements, this method will wait for the 'load' event.
+	 * Returns the numnber of results found under each cateogry.
 	 *
-	 * @param {SupportResultType} category Type of support result item shown.
-	 * @returns {Promise<Locator>} Locator that matches the given selector.
-	 */
-	async getResults( category: SupportResultType ): Promise< Locator > {
-		await this.waitForQueryComplete();
-
-		const selector = selectors.results(
-			category === 'article' ? selectors.supportCategory : selectors.whereCategory
-		);
-		await this.page.waitForSelector( selector );
-		return this.page.locator( selector );
-	}
-
-	/**
-	 * Checks whether popover search shows no results as expected.
+	 * If the category has no results (eg. Show Me Where has no results)
+	 * then the value 0 is returned.
 	 *
-	 * @returns {Promise<void>} No return value.
+	 * @returns {Promise<[number, number]>} Array of numbers.
 	 */
-	async noResultsShown(): Promise< void > {
-		// Note that even for a search query like ;;;ppp;;; that produces no search results,
-		// some links are shown in the popover under the heading `Helpful resources for this section`.
-		await this.page.waitForSelector( selectors.emptyResults );
+	async getResultCount(): Promise< [ number, number ] > {
+		return [
+			await this.page
+				.getByLabel( 'Search Results' )
+				.getByRole( 'list', { name: 'Recommended resources' } )
+				.count(),
+			await this.page
+				.getByLabel( 'Search Results' )
+				.getByRole( 'list', { name: 'Show me where to' } )
+				.count(),
+		];
```

Comment:
It would be easier to follow if we returned the named properties here, e.g.:

```suggestion
		return {
			articleCount: await this.page
				.getByLabel( 'Search Results' )
				.getByRole( 'list', { name: 'Recommended resources' } )
				.count(),
			linkCount: await this.page
				.getByLabel( 'Search Results' )
				.getByRole( 'list', { name: 'Show me where to' } )
				.count(),
		};
```

Reply:
Thanks for the suggestion, makes total sense. 

Incorporated into fix in 23ed24e76e93e4ec1b0598cbedc0e6ad2f89f4dd.